src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -264,16 +264,21 @@ APPLICATION_INFO::FindRequestHandlerAssembly()
             goto Finished;
         }
 
-        if (FAILED(hr = FindNativeAssemblyFromHostfxr(&struFileName)))
+        if (m_pConfiguration->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
         {
-            // TODO eventually make this fail for in process loading.
-            hr = FindNativeAssemblyFromGlobalLocation(&struFileName);
-            if (FAILED(hr))
+            if (FAILED(hr = FindNativeAssemblyFromHostfxr(&struFileName)))
             {
                 goto Finished;
             }
         }
-
+        else
```

Comment:
We don't want to block the option of deploying the new aspnetcorerh.dll at runtime store which contains new features or changes for out of process. It also blocks customer's aspnetcorerh.dll. I would image we should try hostfxr first and then global for out of process

Reply:
You can still install out of process changes globally. Are you suggesting to allow for this with out of process?